go/vt/schemadiff/table.go

```diff
@@ -1346,8 +1346,7 @@ func (c *CreateTableEntity) apply(diff *AlterTableEntityDiff) error {
 					return errors.Wrap(ErrApplyDuplicateKey, keyName)
 				}
 			}
-			for _, col := range opt.IndexDefinition.Columns {
-				colName := col.Column.String()
+			for colName := range getKeyColumnNames(opt.IndexDefinition) {
```

Comment:
This is a separate bug fix to ensure that we can apply an `ALTER` operation on a schema for a functional index. This uses the same logic as we use in the diff computation already.